### ISSUE TITLE
[Sourcing][High Priority][x]‘fdp-date-picker’ doesn’t do the correct date formatting validation based on the different local

### ISSUE THREAD

**Issue by OrangeLi514** [SAP/fundamental-ngx#9879]
#### Is this a bug, enhancement, or feature request?
bug

#### Briefly describe your proposal.
If the locale is Brazil or german (not English locale), the validation for date formatting is not correct. 
~~scenario1- Create a Date type term and choose the date like'16.5.2023' (the day is more than 12). If I manually change the year from 2023 to 2022, the validation is triggered and the error is displayed. That's not correct, because the date is a valid date.~~ working fine as of 0.42.0-rc.31
scenario2-Create a Date type term and choose the date like'3.5.2023' (the day and month are less than 12. If I manually change the month from 3 to 4, the validation is not triggered. However the calendar automatically switch the day and the month. That's not correct.  This only happens specifically when the user omits the "0" in the date. So it works as expected if the user types "03.05.2023" but we see this problem if they simply type "3.5.2023"

https://github.com/SAP/fundamental-ngx/assets/110207830/a9c21dd5-1879-4eee-86c7-86d810a96ca7

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
"@fundamental-ngx/core": "0.40.2-rc.12"
 "@fundamental-ngx/platform": "0.40.2-rc.12",
#### If this is a bug, please provide steps for reproducing it.
please see the attachment

**Comment by OrangeLi514** [comment]
No, we didn't use DayJS datetime adapter. we are using DatetimeAdapter in the code. Are you suggesting using DayJS datetime adapter instead?

**Comment by mikerodonnell89** [comment]
Yes

**Comment by fkolar** [comment]
@OrangeLi514  can you update this ticket with exact setup you have for your DataPicker?

**Comment by fkolar** [comment]
I could not reproduce this issue. 
1. I tried different formats like:
` dateInput: 'DD.MM.YYYY'` or ` dateInput: 'D.M.YYYY'`

2. Logpoint breakpoint into validation method to see if its called
<img width="890" alt="Screenshot 2023-06-26 at 12 01 42" src="https://github.com/SAP/fundamental-ngx/assets/17149942/7e00581c-0a2c-4395-a8d2-06fd483ff8ca">

3. Validation was always triggered, except cases where the dates did not changed. 

` 03.04.2023 `->  `03.4.2023 `-> [here it did not triggered] ->` 03.04.2023`

In the cases described above 
` 3.5.2023` -> ` 4.5.2023` -> it was triggered. 

Used this sample to test using Day.js , where I imported _pt_ locale
<img width="711" alt="Screenshot 2023-06-26 at 12 04 33" src="https://github.com/SAP/fundamental-ngx/assets/17149942/06e06549-d67b-4253-8087-a353fd2a4660">


Closing both for not enough information and what I could tests it worked well. We can re-open once you provide reproducible example. Please follow above recommendation and use Day.js.

**Comment by OrangeLi514** [comment]
Hi Frank, I followed Mike's suggestion to use DayJS. The validation issue is solved, but the calendar switch is not resolved. Please the see the attached screen recording. Mike is aware of this issue and he is fixing this one.
